crates/ra_project_model/src/lib.rs

```diff
@@ -24,6 +25,34 @@ pub use crate::{
     sysroot::Sysroot,
 };
 
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum InlayHintDisplayType {
+    Off,
+    TypeHints,
+    ParameterHints,
+    Full,
+}
+
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct InlayHintOptions {
+    pub display_type: InlayHintDisplayType,
```

Comment:
https://github.com/rust-analyzer/rust-analyzer/issues/2741 have plans to add more various inlay hint types later.

I think we'd better allow to have various combinations of the types turned on simulataneously, since there will be requests for those.
So should we go with a set of `InlayKind`'s instead of this enum?

Reply:
A bitset I guess?

Reply:
Imo, just the `HashSet<InlayKind>` is enough.
Or its `fx` counterpart, since we use `FxHashMap` in the project.

Reply:
I recall @matklad once said it's better not to use HashSet where Vec is enough

Reply:
I'm still a bit of a beginner with rust, but @matklad suggested it be a POD (I'm assuming meaning that it implements `Copy`?). If I use a Vec or any of these structures it can't be a Copy type.